content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx

```diff
@@ -7,18 +7,30 @@ metaDescription: Prisma CLI features that are currently in preview.
 
 To use a Preview CLI feature, [use the `--preview-feature` flag](./) when you run the command. [Share your feedback on all Preview features on GitHub](https://github.com/prisma/prisma/issues/3108).
 
-<Admonition type="info">
+</TopBlock>
 
-There are currently no Preview CLI features.
+## <inlinecode> prisma migrate diff </inlinecode>
 
-</Admonition>
+This command examines two database schema sources to create a migration taking the first to the state of the second.
+The supported types of sources are:
 
-</TopBlock>
+- live databases
+- migration histories
+- Prisma data models
+- an empty schema
+
+The format of the command is:
+
+```terminal
+npx prisma migrate diff --preview-feature --from-empty --to-url $PROD_DATABASE
```

Comment:
Yes, there are a lot more options. The command has an extensive help output. I was not sure whether to recreate that here.

Reply:
```
🔍 Compares the database schema from two arbitrary sources, and outputs the differences either as a human-readable summary (by default) or an executable script.

WARNING prisma migrate diff is currently in Preview (https://pris.ly/d/preview).
There may be bugs and it's not recommended to use it in production environments.

prisma migrate diff is a read-only command that does not write to your datasource(s).
prisma db execute can be used to execute its --script output.

The command takes a source --from-... and a destination --to-....
The source and destination must use the same provider,
e.g. a diff using 2 different providers like PostgreSQL and SQLite is not supported.

It compares the source with the destination to generate a diff. 
The diff can be interpreted as generating a migration that brings the source schema (from) to the shape of the destination schema (to).
The default output is a human readable diff, it can be rendered as SQL using `--script` on SQL databases.

See the documentation for more information https://pris.ly/d/migrate-diff

Usage

  $ prisma migrate diff --preview-feature [options]

Options

                                       -h, --help  Display this help message

From and To inputs (1 `--from-...` and 1 `--to-...` must be provided):
                            --from-url / --to-url  A datasource URL
                        --from-empty / --to-empty  Flag to assume from or to is an empty datamodel
  --from-schema-datamodel / --to-schema-datamodel  Path to a Prisma schema file, uses the datamodel for the diff
--from-schema-datasource / --to-schema-datasource  Path to a Prisma schema file, uses the datasource url for the diff
              --from-migrations / --to-migrations  Path to the Prisma Migrate migrations directory

Shadow database (only required if using --from-migrations or --to-migrations):
                            --shadow-database-url  URL for the shadow database

Output format:
                                         --script  Render a SQL script to stdout instead of the default human readable summary (not supported on MongoDB)

Flag

                                --preview-feature  Run Preview Prisma commands

Examples
 
  From database to database as summary
    e.g. compare two live databases
  $ prisma migrate diff \
    --preview-feature \
    --from-url "$DATABASE_URL" \
    --to-url "postgresql://login:password@localhost:5432/db2"
  
  From a live database to a Prisma datamodel
    e.g. roll forward after a migration failed in the middle
  $ prisma migrate diff \
    --preview-feature \
    --shadow-database-url "$SHADOW_DB" \
    --from-url "$PROD_DB" \
    --to-schema-datamodel=next_datamodel.prisma \
    --script
  
  From a live database to a datamodel 
    e.g. roll backward after a migration failed in the middle
  $ prisma migrate diff \
    --preview-feature \
    --shadow-database-url "$SHADOW_DB" \
    --from-url "$PROD_DB" \
    --to-schema-datamodel=previous_datamodel.prisma \
    --script
  
  From a Prisma Migrate `migrations` directory to another database
    e.g. generate a migration for a hotfix already applied on production
  $ prisma migrate diff \
    --preview-feature \
    --shadow-database-url "$SHADOW_DB" \
    --from-migrations ./migrations \
    --to-url "$PROD_DB" \
    --script

  Execute the --script output with `prisma db execute` using bash pipe `|`
  $ prisma migrate diff \
    --preview-feature \
    --from-[...] \
    --to-[...]  \
    --script | prisma db execute --preview-feature --stdin --url="$DATABASE_URL"
```

Reply:
Thanks, would be good to have this in eventually but given time constraints I've just added in instructions on how to get this help message for now